src/v1_0/rw/ctrl.cpp

```diff
@@ -12,4 +12,20 @@ namespace abb :: rws :: v1_0 :: rw :: ctrl
 
         client.httpPost(uri.str(), content.str(), {Poco::Net::HTTPResponse::HTTP_OK});
     }
-}
\ No newline at end of file
+
+    SafetyViolationInfo getSafetyViolationInfo(RWSClient& client)
+    {
+        std::stringstream uri;
+        uri << Services::CTRL << "/safety/violation";
+
+        RWSResult rws_result = parseXml(client.httpGet(uri.str()).content());
+        
+        SafetyViolationInfo result;
+        result.unsynchronized = std::stoi(xmlFindTextContent(rws_result, XMLAttribute(Identifiers::CLASS, "unsynchronized"))) != 0;
+        result.toolPosViolation = std::stoi(xmlFindTextContent(rws_result, XMLAttribute(Identifiers::CLASS, "tool-pos-violation-status")))!= 0;
+        result.armViolation = std::stoi(xmlFindTextContent(rws_result, XMLAttribute(Identifiers::CLASS, "upper-arm-violation-status"))) != 0;
+        result.axisRangeViolation = std::stoi(xmlFindTextContent(rws_result, XMLAttribute(Identifiers::CLASS, "axis-range-violation-status"))) != 0;
```

Comment:
There is one more parameter in the XML structure returned by the service the `tool-speed-violation-status`.
If we are parsing those information already wouldn't we want to process this information also?
I'm not sure what is this parameter responsible for.